src/bo4e/bo/messlokation.py

```diff
@@ -0,0 +1,132 @@
+"""
+Contains Messlokation class
+and corresponding marshmallow schema for de-/serialization
+"""
+import re
+from typing import List
+
+import attr
+from iso3166 import countries
+from marshmallow import fields
+from marshmallow_enum import EnumField
+
+from bo4e.bo.geschaeftsobjekt import Geschaeftsobjekt, GeschaeftsobjektSchema
+from bo4e.bo.zaehler import Zaehler, ZaehlerSchema
+from bo4e.com.adresse import Adresse, AdresseSchema
+from bo4e.com.dienstleistung import Dienstleistung, DienstleistungSchema
+from bo4e.com.geokoordinaten import Geokoordinaten, GeokoordinatenSchema
+from bo4e.com.hardware import Hardware, HardwareSchema
+from bo4e.com.katasteradresse import Katasteradresse, KatasteradresseSchema
+from bo4e.enum.botyp import BoTyp
+from bo4e.enum.netzebene import Netzebene
+from bo4e.enum.sparte import Sparte
+
+# Structure of a Messlokations-ID
+# Ländercode nach DIN ISO 3166 (2 Stellen)
+# Verteilnetzbetreiber (6 Stellen)
+# Postleitzahl (5 Stellen)
+# Zählpunktnummer (20 Stellen alphanumerisch)
+# source: https://de.wikipedia.org/wiki/Z%C3%A4hlpunkt#Struktur_der_Z%C3%A4hlpunktbezeichnung
+
+_melo_id_pattern = re.compile(r"^[A-Z]{2}\d{6}\d{5}[A-Z\d]{20}$")
+
+
+# pylint: disable=too-many-instance-attributes, too-few-public-methods
+@attr.s(auto_attribs=True, kw_only=True)
+class Messlokation(Geschaeftsobjekt):
+    """
+    Object containing information about a Messlokation
+    """
+
+    # pylint: disable=unused-argument, no-self-use
+    def _validate_messlokations_id(self, messlokations_id_attribute, value):
+        if not value:
+            raise ValueError("The messlokations_id must not be empty.")
+        if not _melo_id_pattern.match(value):
+            raise ValueError(f"The messlokations_id '{value}' does not match {_melo_id_pattern.pattern}")
+        if not value[0:2] in countries:
+            raise ValueError(f"The country code '{value[0:2]}' is not a valid country code")
+
+    # required attributes
+    bo_typ: BoTyp = attr.ib(default=BoTyp.MESSLOKATION)
+    messlokations_id: str = attr.ib(validator=_validate_messlokations_id)
+    sparte: Sparte
+
+    # optional attributes
+    netzebene_messung: Netzebene = attr.ib(default=None)
+    messgebietnr: str = attr.ib(default=None)
+    geraete: List[Hardware] = attr.ib(default=None)
+    messdienstleistung: List[Dienstleistung] = attr.ib(default=None)
+    messlokationszaehler: List[Zaehler] = attr.ib(default=None)
+
+    # only one of the following three optional codenr attributes can be set
+    grundzustaendiger_msb_codenr: str = attr.ib(default=None)
+    grundzustaendiger_msbim_codenr: str = attr.ib(default=None)
+    grundzustaendiger_mdl_codenr: str = attr.ib(default=None)
```

Comment:
See https://github.com/Hochfrequenz/bo4e-modification-proposals/blob/master/markdown/messlokation_mdl.md

Reply:
Removed `grundzustaendiger_mdl_codenr` in e4050593b38c777e53370e4005c78f07f8f3735e